core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -527,4 +528,27 @@ private[spark] object UIUtils extends Logging {
       origHref
     }
   }
+
+  /**
+   * Remove suspicious characters of user input to prevent Cross-Site scripting (XSS) attacks
+   *
+   * For more information about XSS testing:
+   * https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet and
+   * https://www.owasp.org/index.php/Testing_for_Reflected_Cross_site_scripting_(OTG-INPVAL-001)
+   */
+  def stripXSS(url: String): String = {
+    var strippedXSSUrl = url
+    if (strippedXSSUrl != null) {
+      // Avoid null characters or single quote
+      strippedXSSUrl = strippedXSSUrl.replaceAll("(\r\n|\n|\r|%0D%0A|%0A|%0D|'|%27)", "")
+      strippedXSSUrl = StringEscapeUtils.escapeHtml4(strippedXSSUrl)
+    }
+    strippedXSSUrl
+  }
+
+  def stripXSSMap(url: Array[String]): Array[String] = {
```

Comment:
Not sure what this method is trying to do but it just returns its argument. Do you just mean `url.map(stripXSS)`? why Map, why url as names?